### ISSUE TITLE
There was an error in evaluating the Pre-request Script:Error: [jwtSign] not supported argument

### ISSUE THREAD

**Issue by pawanj-deloitte** [joolfe/postman-util-lib#19]
Postman version: 9.5.0

Pre-req script:
eval( pm.globals.get('pmlib_code') )
var header =
{
"alg": "RS256",
"typ": "JWT"
};
var data =
{
"iss": pm.globals.get('jwt_iss'),
"sub": pm.globals.get('jwt_sub'),
"aud": "wd"
};
console.log('Header: ', header);
console.log('Data: ', data);
console.log('JWT Secret: ', pm.globals.get('jwt_secret'));
signedToken = pmlib.jwtSign(pm.globals.get('jwt_secret'),data,header,300,'RS256');
pm.environment.set('jwt_signed', signedToken);
console.log('Signed and encoded JWT', signedToken);

Console:
'Header: ', { alg: 'RS256', typ: 'JWT' }
'Data: ', { iss: 'xxxx',
  sub: 'xxxx',
  aud: 'xxxx' }
'JWT Secret: ', 'xxxx'

Error: [jwtSign] not supported argument

**Comment by joolfe** [comment]
Hi @pawanj-deloitte,

That error is related with the private key that you are using (the jwk) taking your code as example if instead of a correct jwk you use a string like "hi" you get this error, like this:

```
eval( pm.globals.get('pmlib_code') )
var header =
{
"alg": "RS256",
"typ": "JWT"
};
var data =
{
"iss": pm.globals.get('jwt_iss'),
"sub": pm.globals.get('jwt_sub'),
"aud": "wd"
};
console.log('Header: ', header);
console.log('Data: ', data);
// console.log('JWT Secret: ', pm.globals.get('jwt_secret'));
const jwk = 'hi'
signedToken = pmlib.jwtSign(jwk,data,header,300,'RS256');
pm.environment.set('jwt_signed', signedToken);
console.log('Signed and encoded JWT', signedToken);
```
But if you use a correct jwk key then the same code works, as this:

```
eval( pm.globals.get('pmlib_code') )
var header =
{
"alg": "RS256",
"typ": "JWT"
};
var data =
{
"iss": pm.globals.get('jwt_iss'),
"sub": pm.globals.get('jwt_sub'),
"aud": "wd"
};
console.log('Header: ', header);
console.log('Data: ', data);
// console.log('JWT Secret: ', pm.globals.get('jwt_secret'));
const jwk = {
      "e": "AQAB",
      "n": "7-ql2BxRpd5PEKs9h6oouoLJSK1PZgElYnHtC1EvtBcJ5rsPhRLQ95SFuzMVuPzrRymTx2DAiYrFryNNGvYgGMjwjZZ6gL0FebxGooY9P_voShuzNOg22n9aayqM7U61pmfSsu_0Gm8ZFu0hZa4VT7xn_LFIZnB_BTwG9EZtIwyGdHitMcakvdgOjTTmLqzdSjqie_7IHJrLb9JFPLrYsUDGU3v7U4joKwyZWCbaBRLrhmsAxbKhfGBA-zdSAMVtKTuu4HuQ8gnEmYpdL2wZjLH1EpxZpl_dbojzKqqGqDW7inz1SE1y4A86DE4kIzVWMqltxeMpamGOyBtSKYp5Xw",
      "d": "7yt3myJYDse5AEAFfdoW8h_xcx862NBpqWMzUSXlLV4OQ9LVXO7uHPIWAxGhJSIVloKPTKJaLoOGINtLQYZ9ZtfiXvPVOcO10q7rp5e1NbT10_MbSrH4BQgSZmUrymK92Z8Ev9dP_I4mb-GDeq75ldyXuaQuLdkV92zo7dnLcLRciQsFsBirOOICp0RjGdCgoAZMfM5WIvRJ7XYsXP2qGMEOdyeTo3MGWZIYJGSOLvh1OMsWrrqQmPPuh0qnWj5IccJsWQv8oSNPVah9K0b0Zgv5-anr6ZkR33_vr6nK7SiZNIAH48mOrGVKGEyAZGsgoyrNhSVOg39l5jFwkamHUQ",
      "p": "_kSDQ1oCxkTAUjSfsiZ-lj_ineDSTbcfYEP1WyhnNwfUbVP33QYAGg8f36xL65JKZAo4GaZvsVytaqiOiU5u1FFh0Tm7sRU6qG-YjYM0pynsptpQgh4p9mQ0WoURmXoryzIrqq8TRQA871gql14HVavydUrd1Gz10QJkn5cqhJk",
      "q": "8Y0at7FhEtALdBH-L-m2Y4hI9iecA-Y6FXspsiPN4ILXmdO2V1W7pJ9uj-pBt2CWdZbBptd4w3rHiECHOXDYBsn-AjRDwcPRWnhiL_TM4TSlWtyuTAcRg7b_tSWY8ptkIH9nDHYi5_VtdkIWg74Z3oFxWZYS9SKbURnzJ8AxMLc",
      "dp": "papmC9_9psjstaLr2dsS5zpAoLT20QPbQZGFrxsoBMbpNUJcYHmkX1YzYjyUSqxKI9w0qSGz7E-B_dhM5lFgDJ3pqbSAAeTfDxeGwc0W5UN_mUbs5ZpZpNus8L16thDPr9jEcrUXTQHWsesz9OOiy8L4ekuJNPq5PrDcds67OSE",
      "dq": "SZXf5sf9t5pzcsesdPIw4v94oqyMZc-0MCkWGbO4m0Wg4rbRXKe8gzkimGsX3AJrs6Tal7YajwqPUJ2_1CrhVKgiXo-HeLbrxzDXUJ6174F3QkfqPTUGC-cq3Qd4IahfB1tojC5NEfb9gsI9U6MNBHDQlCcY3V7OjBVostpk4d8",
      "qi": "M76jNmuSeQHB5niL0mhtOazj1gQ_gI69yqoCDKgG9lbb1-C1uQ_bpozc9gTwvG0OZ_iqcJ_vNm_vj3d4-2_g8Njm8KTdzILLSw8v4aV-EGijyHPJtJii91SyZY39ZDT-SaWIXAL55aIF5ZfsBhmpwcXK9iCLjy1z2sIjzo023SU",
      "kty": "RSA",
      "kid": "259337db-7412-45da-ad86-b63c97796588",
      "use": "sig"
    }
signedToken = pmlib.jwtSign(jwk,data,header,300,'RS256');
pm.environment.set('jwt_signed', signedToken);
console.log('Signed and encoded JWT', signedToken);
```

Best Regards.

**Comment by pawanj-deloitte** [comment]
Hi @joolfe,
Thank you for sharing an example with JWK that helped clarify my issue. I was initially using Private key in PEM format but now when I tried after converting my key from PEM to JWK the utility worked. Also, additionally to get this working I had to make a slight change to the jwt_secret and parse it as JSON before feeding it to the jwtSign function.

Example:
var jwk = JSON.parse(pm.globals.get('jwt_secret'));
console.log('JWT Secret: ', jwk);
signedToken = pmlib.jwtSign(jwk,data,header,300,'RS256');

Thank You,
Pawan
